server/routes/api/disease.py

```diff
@@ -32,3 +37,47 @@ def get_node(dcid):
                                    post=False,
                                    has_payload=False)
   return response
+
+
+@bp.route('/diseaseParent/<path:dcid>')
+def get_disease_parent(dcid):
+  """Returns parent node for given a disease node."""
+  # list to store parent node
+  list_parent = []
+  # disease of final parent
+  DISEASE_DCID = "bio/DOID_4"
+  curr_dcid = dcid
+  # dcid of the biggest parent node where iteration stops
+  while (curr_dcid != DISEASE_DCID):
+    node_dcids = dc_service.property_values([curr_dcid],
+                                            "specializationOf").get(
+                                                curr_dcid, [])
+    if not node_dcids:
+      break
+    node_dcid = node_dcids[0]
+    node_names = dc_service.property_values([node_dcid],
+                                            "name").get(node_dcid, [])
+    node_name = node_dcid
+    if node_names:
+      node_name = node_names[0]
+    list_parent.append(DiseaseParent(node_dcid, node_name))
+    curr_dcid = node_dcid
+  # return a list of dcid and name lists
+  return Response(json.dumps(list_parent, cls=DiseaseParentEncoder),
```

Comment:
curious what happens if you do this without the encoder? because I think we json.dumps lists of objects elsewhere in our code without needing to encode

Reply:
Without the encoder, I keep getting an error : `class DiseaseParent isn't JSON serializable`! So I googled the error and most of the people suggested using and encoder class!